### DISCUSSION TITLE
Possible to apply prior sample or gene level weights?

### DISCUSSION THREAD

**Discussion by biobenkj** [theislab/scvelo#473] [Q&A]
Are there readily accessible arguments or functions that allow for application of prior sample or gene level weights before inferring velocity? Perhaps at the nearest neighbor initialization? If not, I'd love to discuss ways that this can be achieved to provide multi-modal prior information to RNA velocity. 

The specific quote from the preprint that got me thinking about this is at the end of the discussion and reads, "...This additional readout [scSLAM-seq] can be easily included into the dynamical model, incorporating varying labeling lengths as additional prior. A further extension would be to couple the single-gene dynamical models to formulate regulatory motifs, which may be inferred by leveraging recent parameter inference techniques for scalable estimation and model selection..." 

Thanks again for such an amazing tool!

**Comment by WeilerP** [comment]
@biobenkj, ATM, there are no functions available in scVelo to leverage multi-modal data. For including them in the model itself, you would have to define a new one and adapt the fitting.
In case you were thinking of something like WNN, you could use the Seurat implementation (and convert back to AnnData afterwards) or try the implementation in `muon` [here](https://muon.readthedocs.io/en/latest/omics/multi.html#weighted-nearest-neighbours).
